src/Symfony/Component/Config/Definition/BaseNode.php

```diff
@@ -258,16 +259,39 @@ public function isDeprecated(): bool
     /**
      * @param string $node The configuration node name
      * @param string $path The path of the node
+     *
+     * @return array{package: string, version: string, message: string}
      */
     public function getDeprecation(string $node, string $path): array
     {
+        if (!$this->deprecation) {
+            throw new LogicException(\sprintf('The node "%s" is not deprecated.', $this->getName()));
+        }
+
         return [
             'package' => $this->deprecation['package'],
             'version' => $this->deprecation['version'],
             'message' => strtr($this->deprecation['message'], ['%node%' => $node, '%path%' => $path]),
         ];
     }
 
+    /**
+     * @internal
+     */
+    public function getDeprecationMessage(?NodeInterface $parent = null): string
+    {
+        if (!$this->deprecation) {
+            throw new LogicException(\sprintf('The node "%s" is not deprecated.', $this->getName()));
+        }
+
+        $message = strtr($this->deprecation['message'], ['%node%' => $this->getName(), '%path%' => ($parent ?? $this->parent ?? $this)->getPath()]);
+        if ($this->deprecation['package'] || $this->deprecation['version']) {
+            $message = \sprintf('Since %s %s: ', $this->deprecation['package'], $this->deprecation['version']).$message;
```

Comment:
Would make more sense.
```suggestion
            $message = \sprintf('Since %s %s: %s', $this->deprecation['package'], $this->deprecation['version'], $message);
```